tests/functional/adapter/snapshot_tests/test_snapshot.py

```diff
@@ -0,0 +1,205 @@
+from typing import Dict, List, Iterable
+
+import pytest
+
+from dbt.tests.util import run_dbt
+
+from tests.functional.adapter import common
+from tests.functional.adapter.snapshot_tests import seeds, snapshots
+
+
+MODEL_FACT_SQL = """
+{{ config(materialized="table") }}
+select * from {{ ref('seed') }}
+where id between 1 and 20
+"""
+
+
+class SnapshotBase:
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        """
+        This seed file contains all records needed for tests, including records which will be inserted after the
+        initial snapshot. This makes it so that Redshift creates the correct size varchar columns. This table
+        will only need to be loaded once at the class level. It will never be altered, hence requires no further
+        setup or teardown.
```

Comment:
we do drop these artifacts from the db at the end of the test run right?

Reply:
Oh, I assumed that the `project` fixture handled that. We don't even specify the schema. So I assumed it was created, yielded, and then cleaned up upon test completion. But I can check.

Reply:
The project is yielded out of the fixture here: https://github.com/dbt-labs/dbt-core/blob/5ddd40885e0502e968fd2633127cbb7a1959a40f/core/dbt/tests/fixtures/project.py#L456. After the yield, the teardown attempts to drop the schema. It points out cases where that command may not work and references this issue to resolve those cases: https://github.com/dbt-labs/dbt-core/issues/5041. I would rather solve for it in general with the teardown in `project()` instead of in this PR. Let me know if you feel differently.